pkg/backup/backup.go

```diff
@@ -154,30 +162,39 @@ func (b *Backup) Run() {
 		var ackchan chan backupNowAck
 		select {
 		case <-time.After(interval):
-		case ackchan = <-b.backupNow:
+		case ackchan = <-bc.backupNow:
 			logrus.Info("received a backup request")
 		}
 
-		rev, err := b.saveSnap(lastSnapRev)
+		bs, err := bc.backupManager.SaveSnap(lastSnapRev)
 		if err != nil {
 			logrus.Errorf("failed to save snapshot: %v", err)
 		}
-		lastSnapRev = rev
+
+		if bs != nil {
+			lastSnapRev = bs.Revision
```

Comment:
@xiang90 I changed `SaveSnap` to return `BackupStatus` and `error`.
And only updates lastSnapRev if `BackupStatus` is not nil.